go/internal/feast/transformation/transformation_service.go

```diff
@@ -0,0 +1,204 @@
+package transformation
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/feast-dev/feast/go/internal/feast/registry"
+	"google.golang.org/protobuf/types/known/timestamppb"
+	"io"
+
+	"github.com/apache/arrow/go/v8/arrow"
+	"github.com/apache/arrow/go/v8/arrow/array"
+	"github.com/apache/arrow/go/v8/arrow/ipc"
+	"github.com/apache/arrow/go/v8/arrow/memory"
+	"github.com/feast-dev/feast/go/internal/feast/model"
+	"github.com/feast-dev/feast/go/internal/feast/onlineserving"
+	"github.com/feast-dev/feast/go/protos/feast/serving"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials/insecure"
+)
+
+type GrpcTransformationService struct {
+	project string
+	conn *grpc.ClientConn
+	client *serving.TransformationServiceClient
+}
+
+func NewGrpcTransformationService(config *registry.RepoConfig, endpoint string) (*GrpcTransformationService, error) {
+	opts := make([]grpc.DialOption, 0)
+	opts = append(opts, grpc.WithDefaultCallOptions(), grpc.WithTransportCredentials(insecure.NewCredentials()))
+
+	conn, err := grpc.Dial(endpoint, opts...)
+	if err != nil {
+		return nil, err
+	}
+	client := serving.NewTransformationServiceClient(conn)
+	return &GrpcTransformationService{ config.Project, conn, &client }, nil
+}
+
+func (s *GrpcTransformationService) Close() error {
+	return s.conn.Close()
+}
+
+func (s *GrpcTransformationService) GetTransformation(
+	ctx context.Context,
+	featureView *model.OnDemandFeatureView,
+	retrievedFeatures map[string]arrow.Array,
+	requestContext map[string]arrow.Array,
+	numRows int,
+	fullFeatureNames bool,
+) ([]*onlineserving.FeatureVector, error) {
+	var err error
+
+	inputFields := make([]arrow.Field, 0)
+	inputColumns := make([]arrow.Array, 0)
+	for name, arr := range retrievedFeatures {
+		inputFields = append(inputFields, arrow.Field{Name: name, Type: arr.DataType()})
+		inputColumns = append(inputColumns, arr)
+	}
+	for name, arr := range requestContext {
+		inputFields = append(inputFields, arrow.Field{Name: name, Type: arr.DataType()})
+		inputColumns = append(inputColumns, arr)
+	}
+
+  inputSchema := arrow.NewSchema(inputFields, nil)
```

Comment:
Minor: the indentation on this line and a few others looks strange.

Reply:
should be resolved